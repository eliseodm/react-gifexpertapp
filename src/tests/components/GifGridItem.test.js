import React from 'react';
import { shallow } from 'enzyme';
import "@testing-library/jest-dom";
import { GifGridItem } from '../../components/GifGridItem';

describe('Prueba del componente GifGridItem', () => {
    
            const title = "Soy un Titulo";
            const url = "http://localhost/algo.jpg";
    
            const wrapper = shallow(<GifGridItem title={ title } url={ url } />);

    test('Debe mostrar el componente correctamente', () => {

        expect( wrapper ).toMatchSnapshot();
 
    })
    
    test('<debe de tener un parrafo con el title', () => {

        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
        
    })

    test('debe de obtener la imagen igual al url y alt de los props', () => {
        
        const img = wrapper.find('img');
        //console.log( img.props('img') );

        expect( img.prop('src')).toBe(url);
        expect( img.prop('alt')).toBe(title);
    })

    test('debe tener animate__fadeIn', () => {

    const div = wrapper.find('div');

    //expect( div.prop('className').split(" ").slice(2, 3).toString() ).toBe( 'animate__fadeIn' );
    expect( div.prop('className').includes('animate__fadeIn')).toBe( true );

    })
    
    
    
    
})

