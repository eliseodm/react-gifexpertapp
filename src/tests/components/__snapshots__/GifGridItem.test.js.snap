// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba del componente GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn animate__delay-2s"
>
  <img
    alt="Soy un Titulo"
    src="http://localhost/algo.jpg"
  />
  <p>
     
    Soy un Titulo
     
  </p>
</div>
`;
